LanguageFeatures/Augmentation-libraries/augmenting_constructors_A07_t01_lib.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - A non-redirecting constructor augments a constructor which is not
+///   potentially non-redirecting.
+///
+/// @description Checks that it is a compile-time error a non-redirecting
+/// constructor augments a constructor which is not potentially non-redirecting.
```

Comment:
There would be a `factory` case here, as well. Do we already test in some other test that it is an error to have a non-redirecting factory constructor which is augmented by a redirecting one, and vice versa?

Reply:
Ups! Missed this. Added, thank you for catching this!

Reply:
Got it, thanks!